R/data-survey.R

```diff
@@ -269,3 +288,50 @@ calc_survey_hiv_indicators <- function(survey_meta,
 
   val
 }
+
+
+#' Find Calendar Quarter Midpoint of Two Dates
+#'
+#' @param start_date vector coercibel to Date
+#' @param end_date vector coercibel to Date
+#'
+#' @return A vector of calendar quarters
+#'
+#' @examples
+#' start <- c("2005-04-01", "2010-12-01", "2016-01-01")
+#' end <-c("2005-08-01", "2011-05-01", "2016-06-01")
+#'
+#' @export
+get_mid_calendar_quarter <- function(start_date, end_date) {
+
+  start_date <- lubridate::decimal_date(as.Date(start_date))
+  end_date <- lubridate::decimal_date(as.Date(end_date))
+
+  stopifnot(start_date <= end_date)
+  stopifnot(!is.na(start_date))
+  stopifnot(!is.na(end_date))
+  
+  date4 <- (start_date + end_date) / 2
+  date4 <- round(4 * date4) - 1
```

Comment:
I'd suggest: 

```
year <- floor(date4) 
quarter <- floor((date4 %% 1) * 4) + 1
```

rater than the manipulations in the last two lines here

Reply:
done in e951736.